cellmincer/util/features.py

```diff
@@ -264,19 +264,19 @@ def __init__(
         self.dtype = dtype
         
         # containers
-        self.active_mask_t = active_mask if active_mask is not None else self.determine_active_t_range(ws_base, select_active_t_range)
+        self.active_mask_t = active_mask if active_mask is not None else self.determine_active_t_range(movie_txy, infer_active_t_range)
         self.features = OptopatchGlobalFeatureContainer()
         
         # populate features
         self._populate_features()
         
     @staticmethod
     def determine_active_t_range(
-            ws_base: 'OptopatchBaseWorkspace',
-            select_active_t_range: bool):
+            movie_txy: np.ndarray,
+            infer_active_t_range: bool):
         
-        if select_active_t_range:
-            m_t = np.mean(ws_base.movie_txy, axis=(-1, -2))
+        if infer_active_t_range:
```

Comment:
Not related to this PR, but I noticed so let's fix it: Please take the "magic numbers" down below (like smoothing window, number of histogram bins) and make them method kwargs with default values.

Reply:
Also, you may want to have two strategies for detecting the activation threshold, Otsu and prominence.